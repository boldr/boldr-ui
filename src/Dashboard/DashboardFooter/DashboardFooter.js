/* @flow */
import React from 'react';
import Link from 'react-router-dom/Link';
import classnames from 'classnames';
import { Col, Row } from '../../Layout';
import { StyleClasses } from '../../theme/styleClasses';

const BASE_ELEMENT = StyleClasses.DASHBOARD_FOOTER;
type Props = {
  className: ?string,
  copyright: ?string,
};

const DashboardFooter = (props: Props) => {
  const classes = classnames(BASE_ELEMENT, props.className);
  return (
    <footer className={classes}>
      <Row>
        <Col sm={8} md={9}>
          <span className={`${BASE_ELEMENT}-copyright`}>
            {props.copyright}
          </span>
        </Col>
        <Col sm={4} md={3}>
          <span className={`${BASE_ELEMENT}-powered`}>
            Powered by <a href="https://boldr.io">Boldr</a>
          </span>
        </Col>
      </Row>
    </footer>
  );
};

export default DashboardFooter;
