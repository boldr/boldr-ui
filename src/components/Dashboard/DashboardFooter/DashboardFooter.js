/* @flow */
import React from 'react';
import Link from 'react-router-dom/Link';
import classnames from 'classnames';
import { Grid, Col, Row } from '../../Layout';
import { StyleClasses } from '../../../theme/styleClasses';

const BASE_ELEMENT = StyleClasses.DASHBOARD_FOOTER;
type Props = {
  className: ?string,
  copyright: ?string,
};

const DashboardFooter = (props: Props) => {
  const classes = classnames(BASE_ELEMENT, props.className);
  return (
    <footer className={classes}>
      <Grid fluid>
        <Row>
          <Col xs={12} md={8}>
            <span className={`${BASE_ELEMENT}-copyright`}>
              {props.copyright}
            </span>
          </Col>
          <Col xs={12} md={4}>
            <span className={`${BASE_ELEMENT}-powered`}>
              Powered by <a href="https://boldr.io">Boldr</a>
            </span>
          </Col>
        </Row>
      </Grid>
    </footer>
  );
};

export default DashboardFooter;
