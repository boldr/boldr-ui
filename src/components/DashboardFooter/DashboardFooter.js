/* @flow */
import React from 'react';
import Link from 'react-router-dom/Link';
import classnames from 'classnames';

import { StyleClasses } from '../../theme/styleClasses';

const BASE_ELEMENT = StyleClasses.DASHBOARD_FOOTER;
type Props = {
  className: ?string,
  copyright: ?string,
};

const DashboardFooter = (props: Props) => {
  const classes = classnames(BASE_ELEMENT, props.className);
  return (
    <footer className={ classes }>
      <span className={ `${BASE_ELEMENT}-copyright` }>{ props.copyright }</span>
      <span className={ `${BASE_ELEMENT}-powered` }>Powered by <a href="https://boldr.io">Boldr</a></span>
    </footer>
  );
};

export default DashboardFooter;
