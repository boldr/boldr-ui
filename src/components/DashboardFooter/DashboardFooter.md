# Dashboard Footer

Basic footer element for usage in a dashboard.

```javascript
type Props = {
  className: ?string,
  copyright: ?string,
};
```
