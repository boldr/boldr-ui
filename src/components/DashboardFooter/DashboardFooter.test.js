import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import DashboardFooter from './DashboardFooter';

describe('<DashboardFooter />', () => {
  const wrapper = shallow(<DashboardFooter copyright="widget" />);
  it('renders <DashboardFooter /> without breaking', () => {
    expect(wrapper.find('footer').length).toBe(1);
  });
  it('renders snapshot', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
