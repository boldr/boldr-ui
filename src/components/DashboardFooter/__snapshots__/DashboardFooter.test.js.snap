// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashboardFooter /> renders snapshot 1`] = `
<footer
  className="boldrui-dash-footer"
>
  <Grid
    componentClass="div"
    fluid={true}
  >
    <Row
      componentClass="div"
    >
      <Col
        componentClass="div"
        md={8}
        xs={12}
      >
        <span
          className="boldrui-dash-footer-copyright"
        >
          widget
        </span>
      </Col>
      <Col
        componentClass="div"
        md={4}
        xs={12}
      >
        <span
          className="boldrui-dash-footer-powered"
        >
          Powered by 
          <a
            href="https://boldr.io"
          >
            Boldr
          </a>
        </span>
      </Col>
    </Row>
  </Grid>
</footer>
`;
